packages/client/src/Call.ts

```diff
@@ -766,38 +750,44 @@ export class Call {
     });
 
     // handlers for connection online/offline events
-    // Note: window.addEventListener is not available in React Native, hence the check
-    if (typeof window !== 'undefined' && window.addEventListener) {
-      const handleOnOffline = () => {
-        window.removeEventListener('offline', handleOnOffline);
-        this.logger('warn', '[Rejoin]: Going offline...');
-        this.state.setCallingState(CallingState.OFFLINE);
-      };
-
-      const handleOnOnline = () => {
-        window.removeEventListener('online', handleOnOnline);
-        if (this.state.callingState === CallingState.OFFLINE) {
-          this.logger('info', '[Rejoin]: Going online...');
-          rejoin().catch((err) => {
-            this.logger(
-              'error',
-              `[Rejoin]: Rejoin failed for ${this.reconnectAttempts} times. Giving up.`,
-              err,
-            );
-            this.state.setCallingState(CallingState.RECONNECTING_FAILED);
-          });
+    const unsubscribeOnlineEvent = this.streamClient.on(
+      'connection.changed',
+      (e) => {
+        if (e.type !== 'connection.changed') return;
+        if (!e.online) return;
+        unsubscribeOnlineEvent();
+        if (
+          [CallingState.JOINED, CallingState.JOINING].includes(
+            this.state.callingState,
+          )
+        ) {
```

Comment:
I'm not sure about this... shouldn't we run this code only when the call state is `offline`?

Reply:
In react native, the sfu close event comes very fast when going offline (airplane mode).. so rejoin starts before we can inform to call.ts that we are offline. Hence the state changes from "offline" to ""reconnecting_failed"

Thats why I had to do this instead. 

Alternatively we could choose to not set "reconnecting_failed" when offline

what is better?

Reply:
What is the SFU "close" code in this case? Reconnection should start only if the WS is terminated abruptly (1006).

Reply:
As discussed, I have changed this condition to 

```
        if (
          callingState === CallingState.OFFLINE ||
          callingState === CallingState.RECONNECTING_FAILED
        ) { rejoin()... }
```